src/frequenz/sdk/timeseries/_moving_window.py

```diff
@@ -177,12 +184,43 @@ async def _run_impl(self) -> None:
                 else:
                     self._buffer.update(sample)
 
+                self.count_samples += 1
+                if self.wait_for_num_samples != 0:
+                    if self.count_samples % self.wait_for_num_samples == 0:
+                        self.wait_for_samples_event.set()
+
         except asyncio.CancelledError:
             _logger.info("MovingWindow task has been cancelled.")
             raise
 
         _logger.error("Channel has been closed")
 
+    async def wait_for_samples(self, num_samples: int) -> None:
+        """Wait until a given number of samples has been received.
+
+        The sample counter is updated irrespective of whether this
+        method is called or not. Thus this might trigger when
+        a smaller number of samples than the given number has been
+        updated.
```

Comment:
I guess you decided to go with modulo for a reason, but it is not clear at all for me why. Was this designed with some use case in mind? Maybe you could an example? If feels very strange that if I call `mw.wait_for_samples(10)` it could trigger after receiving only 5 samples or even immediately!

Also, not sure if it will be useful, but given there is this inconsistency between requested number of samples and actual amount of received samples since this method was called, maybe you could return the actual number of samples that were received (à la `read()`)?

Reply:
modulo is gone